rs/execution_environment/src/execution/update.rs

```diff
@@ -447,39 +447,36 @@ impl UpdateHelper {
 
         // Check that the cycles balance does not go below the freezing
         // threshold after applying the Wasm execution state changes.
-        if let Some(state_changes) = &canister_state_changes {
-            let old_balance = self.canister.system_state.balance();
-            let requested = state_changes.system_state_changes.removed_cycles();
-            let reveal_top_up = self
-                .canister
-                .controllers()
-                .contains(&original.call_origin.get_principal());
-            if old_balance < requested + original.freezing_threshold {
-                let err = CanisterOutOfCyclesError {
-                    canister_id: self.canister.canister_id(),
-                    available: old_balance,
-                    requested,
-                    threshold: original.freezing_threshold,
-                    reveal_top_up,
-                };
-                let err = UserError::new(ErrorCode::CanisterOutOfCycles, err);
-                info!(
-                    round.log,
-                    "[DTS] Failed {:?} execution of canister {} due to concurrent cycle change: {:?}.",
-                    original.method,
-                    clean_canister.canister_id(),
-                    err,
-                );
-                self.deallocation_sender.send(Box::new(self.canister));
```

Comment:
Why is this not needed anymore?

Reply:
Good catch. Bad merge conflict resolution. I will add it back (and re-check other places in case I missed something similar).

Reply:
Actually, it's called right below which should also be fine, but let me move it where it was, it's probably better anyway.